lib/queuePopulator/IngestionReader.js

```diff
@@ -202,13 +306,23 @@ class IngestionReader extends LogReader {
     }
 
     _processPublishEntries(batchState, done) {
-        const { entriesToPublish, logRes, logStats, logger } = batchState;
+        const {
+            entriesToPublish, logRes, logStats, logger, initState,
+        } = batchState;
 
-        if (this.remoteLogOffset /* &&
-        config.queuePopulator.logSource.indexOf('ingestion') > -1 */) {
-            batchState.nextLogOffset = this.remoteLogOffset;
-            // return done();
-        } else {
+        // initState.cseq is only fetched at very start of snapshot phase.
+        // We want to save cseq right before we started snapshot
+        // phase to guarantee we don't miss any new entries while snapshot
+        // is in process
+        if (initState.cseq) {
```

Comment:
Is `initState` guaranteed to be an object here? Otherwise, it should be checked first.